File: snippets/rust/src/getting_started.rs
use std::fs;
use std::path::PathBuf;
use std::sync::Arc;

use anyhow::Result;
use bip39::{Language, Mnemonic};
use breez_sdk_liquid::prelude::*;
use log::info;

async fn getting_started() -> Result<Arc<LiquidSdk>> {
    // ANCHOR: init-sdk
    let mnemonic = Mnemonic::generate_in(Language::English, 12)?;

    // Create the default config, providing your Breez API key
    let mut config = LiquidSdk::default_config(
        LiquidNetwork::Mainnet,
        Some("<your-Breez-API-key>".to_string()),
    )?;

    // Customize the config object according to your needs
    config.working_dir = "path to an existing directory".into();

    let connect_request = ConnectRequest {
        config,
        mnemonic: Some(mnemonic.to_string()),
        passphrase: None,
        seed: None,
    };
    let sdk = LiquidSdk::connect(connect_request).await?;
    // ANCHOR_END: init-sdk

    Ok(sdk)
}

async fn getting_started_node_info(sdk: Arc<LiquidSdk>) -> Result<()> {
    // ANCHOR: fetch-balance
    let info = sdk.get_info().await?;
    let balance_sat = info.wallet_info.balance_sat;
    let pending_send_sat = info.wallet_info.pending_send_sat;
    let pending_receive_sat = info.wallet_info.pending_receive_sat;
    // ANCHOR_END: fetch-balance

    Ok(())
}

async fn getting_started_logging(data_dir: String) -> Result<()> {
    // ANCHOR: logging
    let data_dir_path = PathBuf::from(&data_dir);
    fs::create_dir_all(data_dir_path)?;

    LiquidSdk::init_logging(&data_dir, None)?;
    // ANCHOR_END: logging

    Ok(())
}

// ANCHOR: add-event-listener
struct CliEventListener {}
impl EventListener for CliEventListener {
    fn on_event(&self, e: SdkEvent) {
        info!("Received event: {:?}", e);
    }
}

async fn add_event_listener(
    sdk: Arc<LiquidSdk>,
    listener: Box<CliEventListener>,
) -> Result<String> {
    let listener_id = sdk.add_event_listener(listener).await?;
    Ok(listener_id)
}
// ANCHOR_END: add-event-listener

// ANCHOR: remove-event-listener
async fn remove_event_listener(sdk: Arc<LiquidSdk>, listener_id: String) -> Result<()> {
    sdk.remove_event_listener(listener_id).await?;
    Ok(())
}
// ANCHOR_END: remove-event-listener

// ANCHOR: disconnect
async fn disconnect(sdk: Arc<LiquidSdk>) -> Result<()> {
    sdk.disconnect().await?;
    Ok(())
}
// ANCHOR_END: disconnect

File: snippets/go/getting_started.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-liquid-go/breez_sdk_liquid"
)

func Start() (*breez_sdk_liquid.BindingLiquidSdk, error) {
	// ANCHOR: init-sdk
	mnemonic := "<mnemonic words>"

	// Create the default config, providing your Breez API key
	breezApiKey := "<your-Breez-API-key>"
	config, err := breez_sdk_liquid.DefaultConfig(breez_sdk_liquid.LiquidNetworkMainnet, &breezApiKey)
	if err != nil {
		return nil, err
	}

	// Customize the config object according to your needs
	config.WorkingDir = "path to an existing directory"

	connectRequest := breez_sdk_liquid.ConnectRequest{
		Config:   config,
		Mnemonic: mnemonic,
	}

	sdk, err := breez_sdk_liquid.Connect(connectRequest)

	return sdk, err
	// ANCHOR_END: init-sdk
}

func FetchBalance(sdk *breez_sdk_liquid.BindingLiquidSdk) {
	// ANCHOR: fetch-balance
	if info, err := sdk.GetInfo(); err == nil {
		balanceSat := info.WalletInfo.BalanceSat
		pendingSendSat := info.WalletInfo.PendingSendSat
		pendingReceiveSat := info.WalletInfo.PendingReceiveSat
		log.Printf("Balance: %v sats", balanceSat)
		log.Printf("Pending: send %v sats, receive %v sats", pendingSendSat, pendingReceiveSat)
	}
	// ANCHOR_END: fetch-balance
}

// ANCHOR: logging
type SdkLogger struct{}

func (SdkLogger) Log(l breez_sdk_liquid.LogEntry) {
	log.Printf("Received log [%v]: %v", l.Level, l.Line)
}

func SetLogger(logger SdkLogger) {
	breez_sdk_liquid.SetLogger(logger)
}

// ANCHOR_END: logging

// ANCHOR: add-event-listener
type SdkListener struct{}

func (SdkListener) OnEvent(e breez_sdk_liquid.SdkEvent) {
	log.Printf("Received event %#v", e)
}

func AddEventListener(sdk *breez_sdk_liquid.BindingLiquidSdk, listener SdkListener) (string, error) {
	return sdk.AddEventListener(listener)
}

// ANCHOR_END: add-event-listener

// ANCHOR: remove-event-listener
func RemoveEventListener(sdk *breez_sdk_liquid.BindingLiquidSdk, listenerId string) error {
	return sdk.RemoveEventListener(listenerId)
}

// ANCHOR_END: remove-event-listener

// ANCHOR: disconnect
func Disconnect(sdk *breez_sdk_liquid.BindingLiquidSdk) {
	sdk.Disconnect()
}

// ANCHOR_END: disconnect
